build-tools/packages/build-tools/src/common/npmPackage.ts

```diff
@@ -275,11 +277,23 @@ export class Package {
 			return false;
 		}
 		let succeeded = true;
+		// Note that this enumerates peerDeps in addition to deps and devDeps
 		for (const dep of this.combinedDependencies) {
+			if (dep.depClass === "peer") {
+				// Skip peer dependencies because we may legitimately omit them.
+				continue;
+			}
+
+			// TODO: check the installed package semver as well
 			if (
 				!lookUpDirSync(this.directory, (currentDir) => {
-					// TODO: check semver as well
-					return existsSync(path.join(currentDir, "node_modules", dep.name));
+					if (existsSync(path.join(currentDir, "node_modules", dep.name))) {
+						return true;
+					}
+					const alias = dep.version.match(regexNpmAlias)?.groups?.alias;
+					return alias === undefined
+						? false
+						: existsSync(path.join(currentDir, "node_modules", alias));
```

Comment:
The existing logic is correct. pnpm should be placing a directory using the "key" name. The alias aspect should only be impacting the content of that directory.

Reply:
With fresh eyes this is clearly wrong; thanks. I'll keep looking into Pnpm.